includes/compat/class-wc-stripe-email-failed-refund.php

```diff
@@ -0,0 +1,133 @@
+		$this->subject = __( '[{site_title}] Refund failed for #{order_number}.', 'woocommerce-gateway-stripe' );
+	}
+
+	/**
+	 * Get the default e-mail subject.
+	 *
+	 * @return string
+	 */
+	public function get_default_subject() {
+		return $this->subject;
+	}
+
+	/**
+	 * Get the default e-mail heading.
+	 *
+	 * @return string
+	 */
+	public function get_default_heading() {
+		return $this->heading;
+	}
+
+	/**
+	 * Get content html.
+	 *
+	 * @return string
+	 */
+	public function get_content_html() {
+		return wc_get_template_html(
+			$this->template_html,
+			$this->get_template_params(),
+			'',
+			$this->template_base
+		);
+	}
+
+	/**
+	 * Get content plain.
+	 *
+	 * @return string
+	 */
+	public function get_content_plain() {
+		return wc_get_template_html(
+			$this->template_plain,
+			$this->get_template_params(),
+			'',
+			$this->template_base
+		);
+	}
+
+	/**
+	 * Trigger.
+	 *
+	 * @param int $order_id The order ID.
+	 * @param WC_Order|false $order Order object.
+	 */
+	public function trigger( $order_id, $order = false ) {
+		$this->object = $order;
+
+		if ( ! $this->is_enabled() || ! $this->get_recipient() ) {
+			return;
+		}
+
+		$this->find['order-number']    = '{order_number}';
+		$this->replace['order-number'] = $this->object->get_order_number();
+
+		$this->send( $this->get_recipient(), $this->get_subject(), $this->get_content(), $this->get_headers(), $this->get_attachments() );
+	}
+
+	/**
+	 * Returns the refund failure reason in a human-readable form.
+	 *
+	 * @param object $order The order object whose refund request failed.
+	 * @return string
+	 */
+	protected static function get_reason( $order ) {
+		if ( ! is_a( $order, WC_Order::class ) ) {
+			return __( 'Unknown reason', 'woocommerce-gateway-stripe' );
+		}
+
+		$refund_failure_key = $order->get_meta( '_stripe_refund_failure_reason', true );
+		return self::get_refund_reason_description( $refund_failure_key );
+	}
+
+	/**
+	 * Returns the description for a refund reason.
+	 *
+	 * @return string
+	 */
+	protected static function get_refund_reason_description( $refund_reason_key ) {
+		switch ( $refund_reason_key ) {
+			case 'charge_for_pending_refund_disputed':
+				return __( 'Customer disputed the charge while the refund is pending', 'woocommerce-gateway-stripe' );
+			case 'declined':
+				return __( 'Declined by financial partners', 'woocommerce-gateway-stripe' );
+			case 'expired_or_canceled_card':
+				return __( 'Payment method is canceled by the customer or expired by the merchant', 'woocommerce-gateway-stripe' );
+			case 'insufficient_funds':
+				return __( 'Insufficient funds and has crossed the pending refund expiry window', 'woocommerce-gateway-stripe' );
+			case 'lost_or_stolen_card':
+				return __( 'Failed due to loss or theft of the original card', 'woocommerce-gateway-stripe' );
+			case 'merchant_request':
+				return __( 'Failed upon the merchant’s request', 'woocommerce-gateway-stripe' );
```

Comment:
I think we should clean these up in a follow-up PR. I would propose the following wording:
* `charge_for_pending_refund_disputed` - _The charge has been disputed_
* `declined` - _The refund was declined_ - "our financial partners" is the issuing bank/wallet/etc as Stripe's partner
* `expired_or_canceled_card` - _The original payment method has expired or was canceled_
* `insufficient_funds` - _We could not process the refund at this time_ - We should NEVER tell customers that the merchant has insufficient funds. 😬 
* `lost_or_stolen_card` - _The original payment method was lost or stolen_
* `merchant_request` - _We stopped processing the refund_ - I think this should be clearer for users.

Reply:
Thanks for the suggestions, Dale! To speed up things a bit I already applied them in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4403/commits/962820ff90f6f98c8907247109c518d4dc1bf2fe

> We should NEVER tell customers that the merchant has insufficient funds.

Can you elaborate? Since these refund reasons are retrieved from Stripe, the merchant could just look there anyway.

Reply:
That's a good point 👍 . Thank you for elaborating. I think we can leave it like this for now and maybe update the method later if needed to make it more straightforward for merchants.